exec/vector/src/main/codegen/templates/FixedValueVectors.java

```diff
@@ -210,10 +210,16 @@ public void reAlloc() {
     // a zero-length buffer. Instead, just allocate a 256 byte
     // buffer if we start at 0.
 
-    final long newAllocationSize = allocationSizeInBytes == 0
+    long newAllocationSize = allocationSizeInBytes == 0
         ? 256
         : allocationSizeInBytes * 2L;
 
+    // Some operations, such as Value Vector#exchange, can be change DrillBuf data field without corresponding allocation size changes.
+    // Check that the size of the allocation is sufficient to copy the old buffer.
+    while (newAllocationSize < data.capacity()) {
```

Comment:
please also change the following 232 line: `data.setZero(halfNewCapacity, halfNewCapacity)` logic which initializes the allocated fresh buffer content to zero.

Reply:
Done.

Could you clarify why DrillBug#setZero performed only for FixedValueVectors?

Reply:
The change is to keep backward compatible. The reason I guess is that most of the fixed VV is number type. Initializing to zero is to keep the java default value convention.